source/dub/dub.d

```diff
@@ -556,7 +556,10 @@ class Dub {
 			return;
 		}
 
-		foreach (p; versions.byKey) {
+		import std.parallelism : TaskPool, totalCPUs;
+		// Avoid being blocked by a slow download (even on a single-threaded machines)
+        auto taskPool = new TaskPool(totalCPUs.clamp(3, 6));
```

Comment:
You mixed tabs and spaces here.

Also I think this could be `scope` to automatically destroy the TaskPool object at the end.

Should the download speed even be regulated by CPU count at all? I think moving out the parsing/whatever out of the parallel loop would allow us to pick some well working static count of download threads. Then it can submit all downloaded packages for work to some other task pool which will run the more difficult operations on exactly the right count of CPU tasks so systems with many cores can process more packages at once.